types/knex/index.d.ts

```diff
@@ -206,6 +206,14 @@ declare namespace Knex {
         orOn(column1: string, column2: string): JoinClause;
         orOn(column1: string, raw: Raw): JoinClause;
         orOn(column1: string, operator: string, column2: string): JoinClause;
+        onIn(column1: string, values: any[]): JoinClause;
+        onNotIn(column1: string, values: any[]): JoinClause;
+        onNull(column1: string): JoinClause;
+        onNotNull(column1: string): JoinClause;
+        onExists(callback: Function): JoinClause;
```

Comment:
Could this have a better type than `Function`, such as (just guessing) `(result: boolean) => void`?